src/compiler/checker.ts

```diff
@@ -1835,6 +1838,23 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         typeHasCallOrConstructSignatures,
     };
 
+    function getContextualStringLiteralCompletionTypes(expression: Expression) {

```

Comment:
I wanted to move this to the checker since `getCandidateSignaturesForStringLiteralCompletions` is part of the checker.

However, code sharing between the checker and services is quirky at times and now I'm not sure if that was the right decision.